pkg/filter/filter.go

```diff
@@ -29,6 +29,7 @@ type Filterer struct {
 	ChangeInPlace bool
 }
 
+// Filter creates a new Filterer.
```

Comment:
I'd just think to use `type Filter struct` and `func NewFilter`.

Reply:
I'm going to go through and make the API easier to use and more regular, but do it in a holistic way: https://github.com/GoogleCloudPlatform/k8s-cluster-bundle/issues/104